### ISSUE TITLE
Set number of "unique" atom types

### ISSUE THREAD

**Issue by gasevic** [grimme-lab/MindlessGen#75]
Since many different atom types can be introduced in the generation, the outcome can be just a collection of single atoms. Therefore, it would be helpful to restrict the generator to a fixed number of unique atom types so that more atoms have the same atom type.

This could easily be implemented to the `.toml` file. E.g:

`max_num_atoms = 20`
`unique_atom_types = 4`

This could lead to the molecule: C<sub>5</sub>H<sub>10</sub>I<sub>2</sub>Cl<sub>3</sub>

**Comment by marcelmbn** [comment]
Good idea! Should be simple. The only things to keep in mind during implementation:
- contradiction with the `element_composition` (too many different atoms expected)
- contradiction with `max_num_atoms` (if lower than `unique_atom_types`)
- contradiction with `fixed_composition` (`unique_atom_types` would be pointless anyway)
